### ISSUE TITLE
Void artifacts after VQ-VAE reconstruction on Gaussian data

### ISSUE THREAD

**Issue by AresCheah** [nicolasvonluetzow/GaussianGPT#5]
Hi, thanks for the work!

We are currently trying to apply the VQ-VAE to reconstruct the Gaussian data that fitted from the same pipeline as We noticed that after passing our fitted Gaussians through the VQ-VAE and reconstructing them, some scenes sometimes contain white fog-like void artifacts, as shown in the attached example.

The left image is the rendering of our original fitted Gaussians, while the right image is the rendering after VQ-VAE reconstruction, we also tried fine-tuning the VQ-VAE on our own Gaussian data, and these artifacts still appear.

We were wondering: Is this kind of artifact expected or something you have observed during VQ-VAE reconstruction? If not, are there any particular Gaussian attributes or preprocessing steps that are especially important to check when applying the VQ-VAE to custom fitted Gaussians?

Thanks!

<img width="857" height="426" alt="Image" src="https://github.com/user-attachments/assets/358ee8a6-5fb4-474a-a0e0-ea2862d5a203" />

**Comment by nicolasvonluetzow** [comment]
Hey, thanks for your interest in our work!

From the images, I cannot tell exactly whether the issue is caused by holes appearing in the reconstruction or by white fog masking the actual geometry. We observed both failure modes to varying degrees and suspect they are consequences of the 2D supervision.

Our intuition is that holes can arise because the input Gaussians are concentrated on surfaces visible from the training views, creating a biased distribution for the VQ-VAE. Sampling more views when generating/fitting the training data would be a straightforward, though potentially expensive, way to reduce this effect.
We have also observed similar fog-like artifacts, although in our experiments, they were usually dark/black. We suspect the rerendering objective can encourage background-colored Gaussians in poorly constrained regions, which may only become visible from novel viewpoints. In your example, however, the fog appears white, which does not match the black background used during training, so I am not sure whether it is due to the same effect.

Since you are retraining the VQ-VAE on your own data, another thing worth trying could be to explicitly bound the predicted coordinate offsets and scales, as unusually large scales or offsets could contribute to these artifacts. You can add suitable activations or clamps in `internal_representations.py`. I have not tested this myself, though, so I am not sure how much it would help in practice.

Feel free to keep me updated if you manage to narrow down the cause, and happy to take another look if you have additional results or run into anything else!
